client/src/components/reader/Resource/Viewer/Single.js

```diff
@@ -65,38 +69,62 @@ class ResourceViewerSingle extends PureComponent {
     this.props.dispatch(uiReaderActions.setActiveAnnotation(annotationId));
   }
 
-  render() {
+  renderThumbnail() {
     const resource = this.props.resource;
     const variant = "smallLandscape";
     const hasImage = !!get(resource, `attributes.attachmentThumbnails['${variant}']`);
     const height = this.props.height ? this.props.height + 'px' : 'auto';
+
+    return(
+      <div className="resource-preview-overflow"
+           style={{ maxHeight: height }}
+           ref={(r) => { this.single = r; }}
+           onMouseOver={() => { this.setActiveAnnotation(this.props.annotationId); }}
+           onMouseLeave={() => { this.setActiveAnnotation(null); }}
+      >
+        <Resource.Thumbnail
+          key={resource.id}
+          resource={resource}
+          noCrop={hasImage}
+          showTitle
+          showKind={false}
+          variant={variant}
+          additionalClasses="minimal right"
+        />
+      </div>
+    )
+  }
+
+  render() {
     const singleClass = classNames({
       'resource-preview-single': true,
       'transition-out': this.props.fadeIn && !this.state.visible,
       'transition-in': this.props.fadeIn && this.state.visible,
       highlighted: this.props.activeAnnotation === this.props.annotationId
     });
 
+    // Setup link class here
+    const linkClass = classNames({
+      'resource-single-link': true,
+      highlighted: this.props.activeAnnotation === this.props.annotationId
+    });
+
     return (
       <div
         className={singleClass}
         style={{
-          top: this.props.location + 'px',
-          maxHeight: height
+          top: this.props.location + 'px'
         }}
-        ref={(r) => { this.single = r; }}
-        onMouseOver={() => { this.setActiveAnnotation(this.props.annotationId); }}
-        onMouseLeave={() => { this.setActiveAnnotation(null); }}
       >
-        <Resource.Thumbnail
-          key={resource.id}
-          resource={resource}
-          noCrop={hasImage}
-          showTitle
-          showKind={false}
-          variant={variant}
-          additionalClasses="minimal right"
-        />
+        {this.props.authentication && this.props.authentication.authenticated ?
```

Comment:
@naomiyaki This isn't a sufficient check for whether the button should appear or not, since it would allow any authenticated user to see the button. In the future, use the RequireRole component.